### PR TITLE
Get the description from OpenLibrary

### PR DESCRIPTION
**Pull request by fabix**

### CODE REVIEW

File: lib/ui/search_ol_screen/search_ol_screen.dart.dart [hunk 1]
```diff
@@ -287,6 +289,7 @@ class _SearchOLScreenState extends State<SearchOLScreen>
                                   isbn: item.isbn,
                                   olid: item.key,
                                   firstPublishYear: item.firstPublishYear,
+                                  work: "test",
```
Comment (mateusz-bak) [inline lib/ui/search_ol_screen/search_ol_screen.dart.dart:292]: Is that a debug leftover?

File: lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart [hunk 2]
```diff
@@ -17,20 +17,24 @@ class SearchOLEditionsScreen extends StatefulWidget {
     required this.title,
     this.subtitle,
     required this.author,
+    this.description,
```
Comment (mateusz-bak) [inline lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart:20]: Is this needed? I don't see it being used anywhere in the file.

File: lib/ui/add_book_screen/add_book_screen.dart [hunk 3]
```diff
@@ -352,6 +355,17 @@ class _AddBookScreenState extends State<AddBookScreen> {
     });
   }
 
+  void _downloadWork() async {
+    if (widget.work != null) {
+      final OLWork = await OpenLibraryService().getWork(widget.work!);
```
Comment (mateusz-bak) [inline lib/ui/add_book_screen/add_book_screen.dart:360]: please change OLWork to lowerCamelCase like openLibraryWork

File: lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart (hunk 2 above)
Comment (fabix) [reply lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart:20]: This is used l. 109. I've only found this way to transmit the data.

File: lib/ui/search_ol_screen/search_ol_screen.dart.dart (hunk 1 above)
Comment (fabix) [reply lib/ui/search_ol_screen/search_ol_screen.dart.dart]: yes, removed

File: lib/ui/add_book_screen/add_book_screen.dart (hunk 3 above)
Comment (fabix) [reply lib/ui/add_book_screen/add_book_screen.dart]: fixed

File: lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart (hunk 2 above)
Comment (fabix) [reply lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart]: no, removed

**Comment by fabix** [comment]
is it ok now?

File: lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart [hunk 4]
```diff
@@ -192,6 +194,7 @@ class _SearchOLEditionsScreenState extends State<SearchOLEditionsScreen> {
                             onPressed: () => _saveEdition(
                               result: item,
                               cover: item.covers![0],
+                              work: item.works![0].key,
```
Comment (mateusz-bak) [inline lib/ui/search_ol_editions_screen/search_ol_editions_screen.dart:197]: I think this is not null safe. What do you think about:
`work: item.works != null && item.works!.isNotEmpty ? item.works![0].key : null,`

File: lib/ui/search_ol_screen/search_ol_screen.dart.dart [hunk 5]
```diff
@@ -40,6 +40,7 @@ class _SearchOLScreenState extends State<SearchOLScreen>
     required String title,
     String? subtitle,
     required String author,
+    String? description,
```
Comment (mateusz-bak) [inline lib/ui/search_ol_screen/search_ol_screen.dart.dart:43]: I think this is not necessary as description will be downloaded on the AddBookScreen.

File: lib/ui/search_ol_screen/search_ol_screen.dart.dart [hunk 6]
```diff
@@ -50,6 +51,7 @@ class _SearchOLScreenState extends State<SearchOLScreen>
       title: title,
       subtitle: subtitle,
       author: author,
+      description: description,
```
Comment (mateusz-bak) [inline lib/ui/search_ol_screen/search_ol_screen.dart.dart:54]: Same as line 43

**Comment by mateusz-bak** [comment]
Tested, works good. Just please check the review comments.

**Comment by fabix** [comment]
Changes done. To be honest, it's the first time I write in dart (and modify an Android app), so it's far from perfect :smile: ...

**Comment by mateusz-bak** [comment]
It's good, don't worry and thank you for the contributions

**Pull request merged**